gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -185,6 +189,31 @@ fixObject(OMR_VMThread *omrVMThread, MM_HeapRegionDescriptor *region, omrobjectp
 	}
 }
 
+static void
+fixFreeObject(OMR_VMThread *omrVMThread, MM_HeapRegionDescriptor *region, omrobjectptr_t object, void *userData)
+{
+	// OMRPORT_ACCESS_FROM_OMRVMTHREAD(omrVMThread);
+	MM_GCExtensionsBase *extensions = MM_GCExtensionsBase::getExtensions(omrVMThread->_vm);
+	MM_ParallelGlobalGC *collector = (MM_ParallelGlobalGC *)extensions->getGlobalCollector();
+
+	// omrtty_printf("Current object address: %p\n", (void*)object);
+	// omrtty_printf("Current consumed size: %d\n", extensions->objectModel.getConsumedSizeInBytesWithHeader(object));
+	if (extensions->objectModel.isDeadObject(object)) {
+		if (extensions->objectModel.isSingleSlotDeadObject(object)) {
+			((memorySizes*) userData)->freeBytes += extensions->objectModel.getSizeInBytesSingleSlotDeadObject(object);
+		} else {
+			MM_HeapLinkedFreeHeader* freeHeader = MM_HeapLinkedFreeHeader::getHeapLinkedFreeHeader(object);
+			OMRZeroMemory((void*)((uintptr_t)object + sizeof(MM_HeapLinkedFreeHeader)), freeHeader->getSize() - sizeof(MM_HeapLinkedFreeHeader));
+			((memorySizes*) userData)->freeBytes += extensions->objectModel.getSizeInBytesMultiSlotDeadObject(object);
+		}
+	} else {
+		((memorySizes*) userData)->objectBytes += extensions->objectModel.getConsumedSizeInBytesWithHeader(object);
```

Comment:
you can put a comment at the start of this 'else' block that we don't care for marked objects other then being able to assert that we cleared anything this not marked.

Reply:
Do we want to also suggest we don't care about single slot dead object?